node/device/samples/dmpatterns_fwupdate_device.js

```
@@ -0,0 +1,158 @@
+var url = require('url');
+var async = require('async');
+
+// receive the IoT Hub device connection string as a command line parameter
+if(process.argv.length < 3) {
+    console.error('Usage: node dmpatterns_fwupdate_device.js <<IoT Hub Device Connection String>>');
+    process.exit(1);
+}
+
+var connectionString = process.argv[2];
+var client = Client.fromConnectionString(connectionString, Protocol);
+
+// The handler for the firmwareUpdate method.  This initiates
+// the multi-phase firmware update sequence.
+function onFirmwareUpdate(request, response) {
+  // Get the firmware image Uri from the body of the method request
+  var fwPackageUri = request.payload.fwPackageUri;
+  var result = url.parse(fwPackageUri);
+  
+  // Ensure that the url is to a secure url
+  if (result.protocol != 'https:') {
+    response.send(400, 'Invalid URL format.  Must use https:// protocol.', function(err) {
+      if (err) console.error('Error sending method response :\n' + err.toString());
+      else console.log('Response to method \'' + request.methodName + '\' sent successfully.');
+    });
+  } else {
+    // Respond the cloud app for the device method
+    response.send(200, 'Firmware update started.', function(err) {
+      if (err) console.error('Error sending method response :\n' + err.toString());
+      else console.log('Response to method \'' + request.methodName + '\' sent successfully.');
+    });
+
+    initiateFirmwareUpdateFlow(fwPackageUri, function(err){
+      if (!err) console.log("Completed firmwareUpdate flow");
+    });
+  }
+}
+
+// Implementation of firmwareUpdate flow
+function initiateFirmwareUpdateFlow(fwPackageUri, callback) {
+  async.waterfall([
+    function (callback) {
+      downloadImage(fwPackageUri, callback);
+    },
+    function (imageData, callback) {
+      applyImage(imageData, callback);
+    }
+  ], function(err) {
+    if (err) console.error('Error : ' + err.message);
+    callback(null);
```

>callback(null); [](start = 4, length = 15)

this swallows the error.  Did you do this intentionally?